Gifski/Intents.swift

```diff
@@ -1,6 +1,247 @@
+	case aspectRatio(width: Int, height: Int)
+
+	var id: String {
+		do {
+			return try JSONEncoder().encode(self).base64EncodedString()
+		} catch {
+			return "0"
+		}
+	}
+	static var defaultQuery = CropEntityQuery()
+	static var typeDisplayRepresentation = TypeDisplayRepresentation(name: "Crop")
+	var displayRepresentation: DisplayRepresentation {
+		DisplayRepresentation(title: .init(stringLiteral: description))
+	}
+
+	var description: String {
+		switch self {
+		case .noCrop:
+			"No crop"
+		case let .aspectRatio(width: width, height: height):
+			"\(width):\(height)"
+		case let .exact(x: x, y: y, width: width, height: height):
+			"\(width)x\(height) at (\(x),\(y))"
+		}
+	}
+
+	static func from(id: String) throws -> Self {
+		guard let data = Data(base64Encoded: id) else {
+			throw DecodingError.dataCorrupted(DecodingError.Context(codingPath: [], debugDescription: "Invalid ID format"))
+		}
+		return try JSONDecoder().decode(Self.self, from: data)
+	}
+	/**
+	 Convert from entity to CropRect which uses `UnitPoint` and `UnitSize` and a top-left origin (`.exact` uses a bottom-left origin)
+	 */
+	func cropRect(forDimensions dimensions: (Int, Int)) throws -> CropRect? {
+		let (width, height) = dimensions
+		switch self {
+		case .noCrop:
+			return nil
+		case let .exact(x: x, y: bottomLeftY, width: cropWidth, height: cropHeight):
+			let topLeftY = height - bottomLeftY - cropHeight
+			let entityRect = CGRect(x: x, y: topLeftY, width: cropWidth, height: cropHeight)
+			let videoRect = CGRect(origin: .zero, size: .init(width: width, height: height))
+			let intersectionRect = videoRect.intersection(entityRect)
+
+			guard intersectionRect.width >= 1,
+				  intersectionRect.height >= 1 else {
+				throw CropOutOfBounds.cropOutOfBounds(entityRect: entityRect, videoRect: videoRect)
+			}
+			return CropRect(
+				x: Double(intersectionRect.x) / Double(width),
+				y: Double(intersectionRect.y) / Double(height),
+				width: Double(intersectionRect.width) / Double(width),
+				height: Double(intersectionRect.height) / Double(height)
+			)
+		case let .aspectRatio(width: aspectWidth, height: aspectHeight):
+			return CropRect.from(aspectWidth: Double(aspectWidth), aspectHeight: Double(aspectHeight), forDimensions: .init(width: width, height: height))
+		}
+	}
+}
+
+struct CropEntityQuery: EntityQuery {
+	func entities(for identifiers: [CropEntity.ID]) async throws -> [CropEntity] {
+		try identifiers.map {
+			try CropEntity.from(id: $0)
+		}
+	}
+	func suggestedEntities() async throws -> [CropEntity] {
+		PickerAspectRatio.presets.map {
+			.aspectRatio(width: $0.width, height: $0.height)
+		} + [.noCrop]
+	}
+}
+
+enum CropIntentMode: String, AppEnum, CaseIterable {
+	static var caseDisplayRepresentations: [Self: DisplayRepresentation] = [
+		.exact: DisplayRepresentation(stringLiteral: "Exact"),
+		.aspectRatio: DisplayRepresentation(stringLiteral: "Aspect Ratio")
+	]
+
+	case exact
+	case aspectRatio
```

Comment:
This should be first everywhere and the default, since users are more likely to want aspect ratio crop than exact.